k8s-operator/apis/v1alpha1/types_connector.go

```diff
@@ -77,6 +92,40 @@ type SubnetRouter struct {
 	Hostname Hostname `json:"hostname,omitempty"`
```

Comment:
the other approach would be to hoist this up a level and allow both ExitNode and SubnetRouter defintions.

Reply:
I considered that https://github.com/tailscale/tailscale/pull/10724#discussion_r1440527726 and opted to have them on each component, so that users who run `kubectl explain` can see the documentation for their different logic on the relevant components. But I also don't feel very strongly about it.

Reply:
Yeah I think the bigger question is whether we want to support both definitions at the same time.

I don't see a reason not to, but having Tags/Hostname at this layer makes it a more annoying migration later if we decide to.

Reply:
If we support multiple, then we will definitely need to have Tags/Hostname at the component level no? (So that the same is not applied to each component)

Reply:
i mean that we would create a single STS per connector that can do multiple things. like an exit node can be a subnet router today.

Reply:
> allow both ExitNode and SubnetRouter definitions

my main reason is that it feels to me that if we allow both then it should be a list of subnet routers (assuming that multiple might be desired in some cases for reasons other than ownership by different teams). but maybe that is actually not the case

Reply:
> i mean that we would create a single STS per connector that can do multiple things. like an exit node can be a subnet router today.

👍🏼